### DISCUSSION TITLE
Will a Coral EdgeTPU work in an M.2 SATA slot?

### DISCUSSION THREAD

**Discussion by shred86** [blakeblackshear/frigate#1786] [Ask A Question] [answered]
My understanding is there’s two types of underlying interfaces for M.2, SATA and PCI-E (NVMe). Will the EdgeTPU work in either of these cases? I’m assuming it will only work on the M.2 PCI-E (NVMe) interface but just wanted to confirm.

**Comment by pdecat** [comment, marked as answer]
Coral Edge TPU m.2 B+M Key requires PCIe so it will not work in SATA only m.2 port, see:
* https://github.com/google-coral/edgetpu/issues/256#issuecomment-873461106
* https://github.com/magic-blue-smoke/Dual-Edge-TPU-Adapter/blob/6c1b9e6abd2654b8f4576a43418da2fd848d611a/README.md#i-have-only-m2-bm-slot-for-ssd-is-there-some-kind-of-adapter-for-coral-dual-edge-tpu-card
* https://github.com/magic-blue-smoke/Dual-Edge-TPU-Adapter/issues/4#issuecomment-898568212

**Comment by shred86** [comment]
It appears there's even some M.2 PCI-E slots that the Coral EdgeTPU will not work with. I just received an AWOW AK41 Mini PC last night that has an M.2 SATA and M.2 PCI-E (NVMe) slot. I have an M.2 SSD installed where the OS resides and I attempted to install the Coral EdgeTPU into the M.2 PCI-E slot, but it's not being detected at all. I did read a comment that someone mentioned some manufacturers will modify M.2 PCI-E slots to only work with storage devices or wifi cards. Not sure how true that is but it appears to be in my case.
